### PR TITLE
Basic contracts

### PR DESCRIPTION
**Pull request by shenj7**

### CODE REVIEW

File: bin/contracts.ml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+(** Foundational definitions *)
+type currency = USD | GBP | JPY | SGD;;
+type day = Int;;
+type month = Int;;
+type year = Int;;
+type date = Date of year * month * day;;
```
Comment (rhit-naikv) [inline bin/contracts.ml:6]: Might want to see if Ocaml has a library that does like Date or something

**Review by rhit-yangtt — COMMENTED** [review]
ur the expert

**Review by rhit-yangtt — APPROVED** [review]
bottom text

**Review by rhit-naikv — APPROVED** [review]
Check comment

File: bin/contracts.ml (hunk 1 above)
Comment (shenj7) [reply bin/contracts.ml:6]: next pr :)

**Pull request merged**